sub-packages/bionemo-geneformer/tests/bionemo/geneformer/scripts/test_pydantic_train.py

```diff
@@ -85,16 +93,79 @@ def test_pretrain_cli_from_ckpt(tmpdir, data_path):
     assert (result_dir / "test-experiment").exists(), "Could not find test experiment directory."
 
 
+@pytest.mark.slow
+def test_pretrain_api(tmpdir, data_path):
+    from bionemo.geneformer.run.main import main
+
+    result_dir = Path(tmpdir.mkdir("results"))
+    config = f"{result_dir}/test_config.yaml"
+
+    args = argparse.Namespace(
+        recipe="geneformer_tiny_test_recipe",
+        dest=config,
+        data_path=str(data_path),
+        result_dir=result_dir,
+        initial_ckpt_path=None,
+    )
+    config = save_recipe(args)
+
+    args = argparse.Namespace(
+        config=config,
+        model_config_cls=ExposedGeneformerPretrainConfig,
+        data_config_cls=GeneformerPretrainingDataConfig,
+        resume_if_exists=False,
+        nsys_profiling=False,
+        nsys_start_step=0,
+        nsys_end_step=None,
+        nsys_ranks=[0],
+        create_checkpoint_callback=True,
+    )
+    main(args)
+    # NOTE this looks a lot like a magic value. But we also could do yaml.loads(config)['experiment_config']['experiment_name']
+    assert (result_dir / "test-experiment").exists(), "Could not find test experiment directory."
+
+
+@pytest.mark.slow
+def test_finetune_api(tmpdir, data_path):
+    from bionemo.geneformer.run.main import main
+
+    """Uses CLI to invoke the entrypoint"""
+    result_dir = Path(tmpdir.mkdir("results"))
+    config = f"{result_dir}/test_config.yaml"
+    checkpoint_path: Path = load("geneformer/10M_240530:2.0")
+
+    args = argparse.Namespace(
+        recipe="finetune_test_recipe",
+        dest=config,
+        data_path=str(data_path),
+        result_dir=result_dir,
+        initial_ckpt_path=str(checkpoint_path),
+    )
+    config = save_recipe(args)
+
+    args = argparse.Namespace(
+        config=config,
+        model_config_cls=ExposedFineTuneSeqLenBioBertConfig,
+        data_config_cls=GeneformerPretrainingDataConfig,
```

Comment:
This is a different way of calling this than what a user would use. Can you use the string import representation instead?